Code/GraphMol/MolOps.cpp

```diff
@@ -1168,5 +1168,146 @@ std::vector<int> hapticBondEndpoints(const Bond *bond) {
   return oats;
 }
 }  // end of namespace details
-};  // end of namespace MolOps
-};  // end of namespace RDKit
+
+namespace details {
+unsigned int addExplicitAttachmentPoint(RWMol &mol, unsigned int atomIdx,
+                                        unsigned int val, bool addAsQuery,
+                                        bool addCoords) {
+  Atom *newAtom = nullptr;
+  if (addAsQuery) {
+    newAtom = new QueryAtom(0);
+    newAtom->setQuery(RDKit::makeAtomNullQuery());
+  } else {
+    newAtom = new Atom(0);
+  }
+  newAtom->setProp(common_properties::_fromAttachPoint, val);
+  bool updateLabel = false;
+  bool takeOwnership = true;
+  auto idx = mol.addAtom(newAtom, updateLabel, takeOwnership);
+  mol.addBond(atomIdx, idx, Bond::SINGLE);
```

Comment:
I may have missed it somewhere else, but do we have to `setNumExplicitHs` on the parent atom?

Reply:
I think it shouldn't be necessary. We do it in our code because we used the explicit Hs as a placeholder to account for the AttachPT property in valence calculations. But RDKit is not doing that, so it shouldn't be necessary to update the explicit Hs.